### DISCUSSION TITLE
Filter on security group.

### DISCUSSION THREAD

**Discussion by DamagedDingo** [Micke-K/IntuneManagement#190] [Ideas]
I find that I very often want to find everywhere in Intune that a security group is currently used. Would it be possible to add a filter so that it only shows stuff where the group is used (either in assignments or exclusions)?

This is mainly used to clean up old pilot groups post implementation. Currently I run this.

```
$commandsToRun = @(
  'Get-MgBetaDeviceManagementDeviceCompliancePolicy'
  'Get-MgDeviceAppManagementMobileApp'
  'Get-MgDeviceAppManagementTargetedManagedAppConfiguration'
  'Get-MgBetaDeviceAppManagementWindowsManagedAppProtection'
  'Get-MgBetaDeviceAppManagementWindowsInformationProtectionPolicy'
  'Get-MgBetaDeviceManagementConfigurationPolicy'
  'Get-MgBetaDeviceManagementDeviceConfiguration'
  'Get-MgBetaDeviceManagementGroupPolicyConfiguration'
  'Get-MgBetaDeviceAppManagementMobileAppConfiguration'
  'Get-MgBetaDeviceManagementDeviceHealthScript'
  'Get-MgBetaDeviceManagementScript'
  'Get-MgBetaDeviceManagementWindowsAutopilotDeploymentProfile'
)

$Results = $commandsToRun.ForEach{
  & $_ -Property * -ExpandProperty Assignments -All | Where-Object { $_.Assignments.ID -like "*$($Group.id)*" }
} 
```

**Comment by Micke-K** [comment]
Hello,

You can you the Intune Assignment Tools view for this (Views -> Intune Tools -> Intune Assignments)

The current version can only filter from an exported folder. Do a Bulk Export of the whole environment and run the Intune Assignment tool.

I will revisit this and hopefully have time to add support for online searches in the Future.

Cheers!

**Reply by DamagedDingo** [reply]
That's exactly what I was hoping to achieve. It would be nice if the Intune Manager showed the assignments as well but this is still a huge time saver. 
If I can wrap my head around the XAML then I will try to make the changes myself.
